jlibra-core/src/main/java/dev/jlibra/admissioncontrol/transaction/Transaction.java

```diff
@@ -1,28 +1,40 @@
 package dev.jlibra.admissioncontrol.transaction;
 
+import static dev.jlibra.serialization.Deserialization.readByteSequence;
+import static dev.jlibra.serialization.Deserialization.readInt;
+import static dev.jlibra.serialization.Deserialization.readLong;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
 import org.immutables.value.Value;
 
 import dev.jlibra.AccountAddress;
+import dev.jlibra.LibraRuntimeException;
 import dev.jlibra.serialization.ByteSequence;
 import dev.jlibra.serialization.LibraSerializable;
 import dev.jlibra.serialization.Serializer;
 
 @Value.Immutable
-public interface Transaction extends LibraSerializable {
+public abstract class Transaction implements LibraSerializable {
```

Comment:
Could we keep this as an interface? New method can still be added either way

Reply:
Haha, I think I suggested just the contrary to @ketola - my point was that there is a semantic "slip" in using an interface and implementing logic in it, especially like this - if I implement the interface, the default methos for serialization is used, defining the serialization strategy for my class. This does not seem legit to me, with an abtract class the problem stays the same, but at least I know that an abstract class defines certain strategies.

Reply:
I suggest we can merge this as it is now, as it is like other similar implementations in the project. Let's create a separate task to refactor the serialization for all classes once we have an idea how to do it.